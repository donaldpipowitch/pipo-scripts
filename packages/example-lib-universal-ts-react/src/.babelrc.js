module.exports = require("pipo-scripts").getBabelConfig();
