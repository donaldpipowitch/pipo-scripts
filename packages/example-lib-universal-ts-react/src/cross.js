"use strict";
exports.__esModule = true;
var react_1 = require("react");
var cross_png_1 = require("./cross.png");
/**
 * Shows a cross by using a `<img />`.
 */
exports.Cross = function () { return react_1["default"].createElement("img", { src: cross_png_1["default"] }); };
