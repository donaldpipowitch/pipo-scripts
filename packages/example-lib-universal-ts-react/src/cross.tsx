import React, { SFC } from 'react';
import crossSrc from './cross.png';

/**
 * Shows a cross by using a `<img />`.
 */
export const Cross: SFC = () => <img src={crossSrc} />;
