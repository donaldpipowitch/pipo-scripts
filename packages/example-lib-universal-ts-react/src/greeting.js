"use strict";
exports.__esModule = true;
var react_1 = require("react");
/**
 * Greets the user.
 */
exports.Greeting = function (_a) {
    var name = _a.name;
    return react_1["default"].createElement("p", null,
        "Hello ",
        name,
        "!");
};
