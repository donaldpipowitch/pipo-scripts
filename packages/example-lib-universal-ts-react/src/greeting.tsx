import React, { SFC } from 'react';

/**
 * The props for `<Greeting />`.
 */
export interface GreetingProps {
  /**
   * The name of the user (e.g. "John Doe" or just "John").
   */
  name: string;
}

/**
 * Greets the user.
 */
export const Greeting: SFC<GreetingProps> = ({ name }) => <p>Hello {name}!</p>;
