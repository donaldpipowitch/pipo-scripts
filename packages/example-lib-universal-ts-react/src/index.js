"use strict";
exports.__esModule = true;
var greeting_1 = require("./greeting");
exports.Greeting = greeting_1.Greeting;
var smile_1 = require("./smile");
exports.Smile = smile_1.Smile;
var cross_1 = require("./cross");
exports.Cross = cross_1.Cross;
