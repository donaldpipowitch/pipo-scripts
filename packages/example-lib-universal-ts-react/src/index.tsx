export { Greeting, GreetingProps } from './greeting';
export { Smile } from './smile';
export { Cross } from './cross';
