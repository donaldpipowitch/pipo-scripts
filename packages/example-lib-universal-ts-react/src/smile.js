"use strict";
exports.__esModule = true;
var react_1 = require("react");
require("./smile.css");
/**
 * Shows a smile by using a background image.
 */
exports.Smile = function () { return react_1["default"].createElement("div", { className: "smile" }); };
