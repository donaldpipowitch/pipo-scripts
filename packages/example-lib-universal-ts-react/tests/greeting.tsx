import React from 'react';
import renderer from 'react-test-renderer';
import { Greeting } from '../src/greeting';

test('Render <Greeting />', () => {
  const component = renderer.create(<Greeting name="John Doe" />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
