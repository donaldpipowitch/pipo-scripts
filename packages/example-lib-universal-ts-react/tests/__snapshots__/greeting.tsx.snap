// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render <Greeting /> 1`] = `
<p>
  Hello 
  John Doe
  !
</p>
`;
